components/automate-chef-io/content/docs/eas.md

```diff
@@ -0,0 +1,108 @@
+Chef Habitat artifact is the binary distribution produced by the Chef Habitat build tools.
+While a very simple application might have only one origin, most modern applications follow common architecture patterns that result in multiple artifacts.
+Artifacts are built according to the instructions in the `plan.sh` or `plan.ps1` file.
+Each artifact contains a software library or application, configuration settings for the application, and lifecycle hooks.
+
+_Channel_
+A channel is a tag that you apply to an artifact.
+By default, when you upload your application artifact to Builder, Chef Habitat labels your application artifact with the `unstable` tag, which places it into the "unstable" channel.
+When you promote your application artifact to another channel, such as "test" or "stable", Chef Habitat applies the new tag and makes it available to the respective channel.
+You can apply more than one tag to a single artifact, making it available for use in multiple channels.
+For more information, see the Chef Habitat [Continuous Deployment Using Channels](https://www.habitat.sh/docs/using-habitat/#continuous-deployment-using-channels) documentation.
+
+_Deployment_
+Each instance of an application downloaded from a channel into an environment is called a deployment.
+
+_Environment_
+Each application lifecycle stage is supported by automation for consistent and reproduceable environments. Examples of environments include "development," "acceptance," and "production". Every environment in your application lifecycle has its own supervisors, service groups, and services.
+
+_Supervisor_
+The supervisor is Chef Habitat's process manager, which means they control the processes related to an application, including starting, monitoring, and updating services.
+Supervisors in each environment subscribe to a channel, which means that they watch for changes to an application artifact's identification string.
+Once a supervisor detects a change to artifact, it deploys the new package into its own environment and updates all of the services for that service group.
+For more information about automated deployments, see the Chef Habitat [update-strategy](https://www.habitat.sh/docs/using-habitat/#update-strategy) documentation.
+
+_Service_
+A service is any individual Chef Habitat package that you use in your application.
+For example, if you have a simple Node.js web application, it would depend on Chef Habitat's `core/node` package.
+When you upload your application's build artifact to Builder and promote it to the 'stable' channel, the superisors then deploy it into the 'production' environment.
+That single running instance of Node in your production environment is called a service.
+
+_Service Group_
+A service group is the collection of your application's services, that are all part of the same artifact, tagged with the same channel, and deployed to the same environment. In some cases, very complex applications may have more than one service group.
```

Comment:
A "service group" is probably best thought of as a kind of handle that lets you can deal with multiple instances of a service running in your "environment" as one thing.

Whether you have one instance of, say, Redis running, or five separate replicas, they would all be running in a single "service group". Commands like `hab config apply` use service group names to target which services get the newly-applied configuration values. This lets you reconfigure those five separate Redis replicas in a single logical operation, rather than having to seek out the individual machines they're running on and apply the configuration to each in turn.

A "service group" also allows you to refer to the collection of services in configuration templates. If you need to know the IP addresses of all those Redis replicas to configure your web app, you get that information by going through the service group.

The service group also defines the domain over which leader elections occur (the leader of those Redis services has to _be_ one of those Redis services, and only those Redis services can vote on which one becomes the leader). I wouldn't draw much attention to that, however, since we're talking about marking that functionality as "beta", because there are some kinks that need to be worked out.

I wouldn't say "tagged with the same channel" for the same reasons I argue that we shouldn't necessarily talk about channels as tags to begin with.

I'd also suggest rewording the "part of the same artifact" bit as well. I suspect the intention is to convey that all the services in a service group are running the same release of that service. "Part of the same artifact" implies some kind of containment relationship that's not really true.

(Also note that while it is the desired and ideal case that all the services in a service group are indeed running the same release of a given service, and that they're all being updated in the same way, from the same channel, and pulling from the same Builder instance, there isn't anything directly in Habitat today that guarantees that. Ideally there would be some "definition" of a Service Group that would exist somewhere "in the network", but that is not currently the case. This is only achieved through consistent management of Supervisors and services, and would probably need to be layered on by Automate or something similar.

If applications are collections of multiple services, then by definition they will _all_ have more than one service group.

Reply:
That is a great definition!  
Sadly, I've boiled it down to two lines, but I'm holding on to this for future use.